mirror_maker.go

```diff
@@ -148,6 +172,14 @@ func (this *MirrorMaker) startConsumers() {
 		if this.config.PreserveOrder {
 			numProducers := this.config.NumProducers
 			config.Strategy = func(_ *Worker, msg *Message, id TaskId) WorkerResult {
+				if this.config.TimingsProducerConfig != "" {
+					if record, ok := msg.DecodedValue.(*avro.GenericRecord); ok {
+						msg.DecodedValue = this.addTiming(record)
+					} else {
+						return NewProcessingFailedResult(id)
```

Comment:
We may want to fail the offset more with this, possibly not sure need to think about it more. Thoughts?